compiler/semstmts.nim

```diff
@@ -531,6 +532,15 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
 
     # this can only happen for errornous var statements:
     if typ == nil: continue
+    let maybeConcrete = skipTypes(typ, {tyGenericInst, tyAlias,
+      tyDistinct, tyRef, tyPtr})
+
+    if efInConcept notin flags and a.sons[length-1].kind == nkEmpty and
+          maybeConcrete.kind notin ConcreteTypes or
```

Comment:
This condition is not particularly accurate. I can check whether a type is concrete with the proc `isMetaType` from ast.nim